domain/src/test/java/org/oppia/domain/topic/StoryProgressControllerTest.kt

```diff
@@ -136,116 +179,249 @@ class StoryProgressControllerTest {
+
+      verify(mockRecordProgressObserver, atLeastOnce()).onChanged(recordProgressResultCaptor.capture())
+      assertThat(recordProgressResultCaptor.value.isSuccess()).isTrue()
+
+      verify(mockTopicProgressObserver, atLeastOnce()).onChanged(topicProgressResultCaptor.capture())
+      assertThat(topicProgressResultCaptor.value.isSuccess()).isTrue()
+
+      val topicProgress = topicProgressResultCaptor.value.getOrThrow()
+      assertThat(topicProgress).isNotNull()
+      assertThat(topicProgress.storyProgressMap[STORY_ID_1]).isNotNull()
+    }
 
   @Test
-  fun testRecordCompletedChapter_validStory_validChapter_notYetCompleted_succeeds() {
-    val recordProgressLiveData = storyProgressController.recordCompletedChapter(TEST_STORY_ID_1, TEST_EXPLORATION_ID_2)
+  @ExperimentalCoroutinesApi
+  fun testRecordCompletedChapter_validData_recordProgress_getTopicProgressForInvalidProfileID_noResultFound() =
+    runBlockingTest(coroutineContext) {
+      storyProgressController.recordCompletedChapter(PROFILE_ID_1, TOPIC_ID_1, STORY_ID_1, EXPLORATION_ID_1)
+        .observeForever(mockRecordProgressObserver)
+      advanceUntilIdle()
 
-    val recordProgressResult = recordProgressLiveData.value
-    assertThat(recordProgressResult).isNotNull()
-    assertThat(recordProgressResult!!.isSuccess()).isTrue()
-  }
+      storyProgressController.getTopicProgress(PROFILE_ID_2, TOPIC_ID_1).observeForever(mockTopicProgressObserver)
+
+      verify(mockRecordProgressObserver, atLeastOnce()).onChanged(recordProgressResultCaptor.capture())
+      assertThat(recordProgressResultCaptor.value.isSuccess()).isTrue()
+
+      verify(mockTopicProgressObserver, atLeastOnce()).onChanged(topicProgressResultCaptor.capture())
+      assertThat(topicProgressResultCaptor.value.isSuccess()).isFalse()
+    }
 
   @Test
-  fun testRecordCompletedChapter_validStory_validChapter_notYetCompleted_marksChapterAsCompleted() {
-    storyProgressController.recordCompletedChapter(TEST_STORY_ID_1, TEST_EXPLORATION_ID_2)
+  @ExperimentalCoroutinesApi
+  fun testRecordCompletedChapter_validData_recordProgress_getTopicProgressForInvalidTopicID_noResultFound() =
+    runBlockingTest(coroutineContext) {
+      storyProgressController.recordCompletedChapter(PROFILE_ID_1, TOPIC_ID_1, STORY_ID_1, EXPLORATION_ID_1)
+        .observeForever(mockRecordProgressObserver)
+      advanceUntilIdle()
 
-    val storyProgress = storyProgressController.getStoryProgress(TEST_STORY_ID_1).value!!.getOrThrow()
-    assertThat(storyProgress.getChapterProgress(1).explorationId).isEqualTo(TEST_EXPLORATION_ID_2)
-    assertThat(storyProgress.getChapterProgress(1).playState).isEqualTo(COMPLETED)
-  }
+      storyProgressController.getTopicProgress(PROFILE_ID_1, TOPIC_ID_2).observeForever(mockTopicProgressObserver)
+
+      verify(mockRecordProgressObserver, atLeastOnce()).onChanged(recordProgressResultCaptor.capture())
```

Comment:
Here & elsewhere: you can shorten this tests if you'd like by moving some of these calls to verification helpers. Note that such helpers should be kept specific to verifying a single logical idea. In this case, I could see two existing:

```kotlin
private fun verifyRecordProgressSucceeded() {
  verify(mockRecordProgressObserver, atLeastOnce()).onChanged(...)
  assertThat(recordProgressResultCaptor.value.isSuccess()).isTrue()
}

private fun verifyGetTopicProgressResultFailed() {
  ...
}
```

These types of helpers keep the high-level details of the test roughly the same, and move unimportant specifics out of the test body (such as interacting with mocks or captors like this).